### PR TITLE
RuboCopBear: Fix CI breakage

### PR DESCRIPTION
**Pull request by yash-nisar**
https://github.com/bbatsov/rubocop/pull/4226 indicates that `--stdin`
takes filename as an argument which was incorporated. Also, the tests
were enhanced that involved the bear to be run on files with the `.rb`
(ruby) extension because initially the bear was run on the generated
temporary files that was the cause of error.

Fixes https://github.com/coala/coala-bears/issues/1548

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: bears/ruby/RuboCopBear.py [hunk 1]
```diff
@@ -36,7 +36,7 @@ class RuboCopBear:
     def create_arguments(filename, file, config_file, rubocop_config: str=''):
         # Need both stdin and filename. Explained in this comment:
         # https://github.com/bbatsov/rubocop/pull/2146#issuecomment-131403694
-        args = (filename, '--stdin', '--format=json')
+        args = ('--stdin', filename, '--format=json')
```
Comment (jayvdb) [inline bears/ruby/RuboCopBear.py]: Is this change compatible with earlier versions of rubocop?

Comment (yash-nisar) [inline bears/ruby/RuboCopBear.py]: Yes :+1: , tested will versions `0.47.1`, `0.48.1`, `0.49.1`

File: tests/ruby/RuboCopBearTest.py [hunk 2]
```diff
@@ -1,57 +1,91 @@
...
+from coalib.results.Result import Result
+from coalib.results.RESULT_SEVERITY import RESULT_SEVERITY
+from coalib.settings.Section import Section
+from coalib.testing.BearTestHelper import generate_skip_decorator
+from coalib.testing.LocalBearTestHelper import LocalBearTestHelper
 
 from bears.ruby.RuboCopBear import RuboCopBear
-from coalib.testing.LocalBearTestHelper import verify_local_bear
 
-good_file = """def good_name
```
Comment (jayvdb) [inline tests/ruby/RuboCopBearTest.py]: why do we need to move the text out into real filenames?

Does rubocop require that there is a valid file with this filename?

Cant `LocalBearTestHelper` create temporary files with an appropriate filename/extension?

If it cant, then writing tests is much harder, which is sort of the anti-thesis of your project.

Comment (yash-nisar) [inline tests/ruby/RuboCopBearTest.py]: Yes, rubocop requires that we have a valid filename extension (`.rb`). Temporary files were created in the previous case when we had to use `verify_local_bear(...)` but without a valid extension. See the commit message.

Comment (jayvdb) [inline tests/ruby/RuboCopBearTest.py]: So, you need to improve `LocalBearTestHelper` :P

Comment (yash-nisar) [inline tests/ruby/RuboCopBearTest.py]: Actually, this is a linter specific problem. This is the first linter that I've come across that is careful about the filename extension. Moreover, it would make less sense IMO to improve `LocalBearTestHelper` to take care of this case only. Also we had to somehow improve the tests and use `check_results(...)` instead of `verify_local_bear(...)`.

**Comment by yash-nisar** [comment]
@Makman2 Tagging you since this PR involved improvisation of tests.

File: bears/ruby/RuboCopBear.py [hunk 3]
```diff
@@ -19,7 +19,7 @@ class RuboCopBear:
     """
 
     LANGUAGES = {'Ruby'}
-    REQUIREMENTS = {GemRequirement('rubocop', '0.47.1'),
+    REQUIREMENTS = {GemRequirement('rubocop'),
```
Comment (Makman2) [inline bears/ruby/RuboCopBear.py]: why did you remove the version specification?

File: tests/ruby/RuboCopBearTest.py [hunk 4]
```diff
@@ -1,57 +1,91 @@
...
+            [Result.from_values('RuboCopBear (Layout/CommentIndentation)',
+                                message='Incorrect indentation detected '
+                                        '(column 2 instead of 1).',
+                                file=get_testfile_path(filename),
+                                line=2,
+                                column=3,
+                                end_line=2,
+                                end_column=20,
+                                severity=RESULT_SEVERITY.INFO)],
+            filename=get_testfile_path(filename),
```
Comment (Makman2) [inline tests/ruby/RuboCopBearTest.py]: you should also check good files :+1:

**Comment by Makman2** [comment]
You could write into the commit message body that part of the fixing was to improve the tests.

File: bears/ruby/RuboCopBear.py (hunk 3 above)
Comment (yash-nisar) [inline bears/ruby/RuboCopBear.py]: Our motive is to use the latest version of rubocop. We were using `0.47.1` in the earlier case and due to an upstream release i.e. `0.48`, our CI broke. As a temporary fix, I had to pin the version to `0.47.1` to make sure that the tree was green again. Now that we have to solution, I think it would be a good option to get rid of the pinning and install the latest version.

File: tests/ruby/RuboCopBearTest.py (hunk 4 above)
Comment (yash-nisar) [inline tests/ruby/RuboCopBearTest.py]: Done :+1:

**Comment by yash-nisar** [comment]
Updated @Makman2

File: bears/ruby/RuboCopBear.py (hunk 3 above)
Comment (Makman2) [inline bears/ruby/RuboCopBear.py]: though I would recommend to fix it currently for 0.48, and upgrade to newer releases explicitly with an issue. But up to you :+1: Not sure actually what our requirements strategy is for bears^^

File: tests/ruby/RuboCopBearTest.py [hunk 5]
```diff
@@ -1,57 +1,111 @@
...
+            settings={'indent_size': 1})
+
+    def test_good_indent_size(self):
+        filename = 'bad_indent_file.rb'
```
Comment (Makman2) [inline tests/ruby/RuboCopBearTest.py]: maybe just call the file `indent.rb`, as in this context it isn't bad?

**Comment by Makman2** [comment]
ack 51507c5

**Comment by Makman2** [comment]
reack d35b120

**Comment by Makman2** [comment]
@rultor merge
